src/Fields/ClassRuleParser.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Statamic\Fields;
+
+use Statamic\Support\Str;
+
+class ClassRuleParser
+{
+    public function parse(string $rule): array
+    {
+        $rule = Str::substr($rule, 4);
+
+        if (! str_contains($rule, '(')) {
+            return [$rule, []];
+        }
+
+        $class = Str::before($rule, '(');
+
+        $arguments = Str::of($rule)
+            ->between('(', ')')
+            ->explode(',')
+            ->mapWithKeys(function ($arg, $key) {
+                $arg = trim($arg);
+
+                if (str_contains($arg, ':')) {
+                    [$key, $arg] = explode(':', $arg);
+                    $key = trim($key);
+                    $arg = trim($arg);
```

Comment:
Couple of things maybe worth considering.

If the $arg contains `:` but its not a named argument I think it will trip up here?
If its a named argument and $arg contains `:` then it'll only take the $arg up to its first : ?

Reply:
Good call. Fixed.